### PR TITLE
Refactor `TransitionAnimationType` and `InteractiveGestureType` to conform to `IBEnum`

### PR DESCRIPTION
**Pull request by JakeLin**
In this PR, I have followed @lastMove 's pattern for `AnimationType` to refactor `TransitionAnimationType` and `InteractiveGestureType` to conform to `IBEnum`. Also, moved all subtype like `Direction` within `TransitionAnimationType`. 

Relate to #221 

@lastMove @tbaranes please have a look when you have time, thanks.

### CODE REVIEW

**Comment by lastMove** [comment]
I will review it tomorrow morning (In ten hours ^^). 
I have almost finish to refactor `PresentationDesignable` I will make a PR tomorrow :)

**Comment by JakeLin** [comment]
@lastMove thanks, no worries, take your time. I will work on activity indicators.

**Comment by JakeLin** [comment]
I have rebased it from `swift3` branch to support Xcode 8 GM. Some of the transition animations have been broken 😱, but I think we can fix them later.

File: IBAnimatable/IBEnum.swift [hunk 1]
```diff
@@ -30,10 +28,10 @@ public extension IBEnum {
    - Returns: A tuple containing the name and an array of parameter string
    */
   static func extractNameAndParams(from string: String) -> (name: String, params: [String]) {
-    let tokens = string.lowercased().components(separatedBy: CharacterSet(charactersIn: "()")).filter({!$0.isEmpty})
+    let tokens = string.lowercased().components(separatedBy: CharacterSet(charactersIn: "()")).filter{ !$0.isEmpty }
     let name = tokens.first ?? ""
     let paramsString = tokens.count >= 2 ? tokens[1] : ""
-    let params = paramsString.components(separatedBy: ",").filter({!$0.isEmpty})
+    let params = paramsString.components(separatedBy: ",").filter{ !$0.isEmpty }.map{ $0.trimmingCharacters(in: .whitespaces) }
```
Comment (lastMove) [inline IBAnimatable/IBEnum.swift:35]: Good Catch 👍

File: IBAnimatable/InteractiveGestureType.swift [hunk 2]
```diff
@@ -6,64 +6,65 @@
...
-      return .default
-    } else if interactiveGestureType.hasPrefix("pan") || interactiveGestureType.hasPrefix("screenEdgePan") ||
-      interactiveGestureType.hasPrefix("pinch") {
-      return fromStringWithDirection(interactiveGestureType)
-    }
-    return nil
+  /// Returns the string value without qualification
+  public var stringValueWithoutQualification: String {
+    let namespace = "IBAnimatable." + String(describing: InteractiveGestureType.self) + "." + String(describing: GestureDirection.self) + "."
+    return String(describing: self).replacingOccurrences(of: namespace, with: "")
```
Comment (lastMove) [inline IBAnimatable/InteractiveGestureType.swift:25]: Very useful; Maybe later we could make a Generic version for all IBEnum.

File: IBAnimatableApp/TransitionTableViewController.swift [hunk 3]
```diff
@@ -76,8 +76,7 @@ private extension TransitionTableViewController {
     transitionAnimationsHeaders.append("Slide")
     transitionAnimations.append(["Slide(Left, fade)", "Slide(Right)", "Slide(Top, fade)", "Slide(Bottom)"])
     transitionAnimationsHeaders.append("Others")
-    transitionAnimations.append(["SystemRotate", "SystemRippleEffect", "SystemSuckEffect", "Explode(10,-10,10)"])
-
+    transitionAnimations.append(["SystemRotate", "SystemRippleEffect", "SystemSuckEffect", "Explode", "Explode(10,-20,20)"])
```
Comment (lastMove) [inline IBAnimatableApp/TransitionTableViewController.swift:79]: It Was "Explode(10,-10,10)". Is this change intentional ?

Comment (JakeLin) [reply IBAnimatableApp/TransitionTableViewController.swift:79]: Yes, the default value is Explode(10, -10,10), same as without params Explode

File: IBAnimatable/InteractiveGestureType.swift (hunk 2 above)
Comment (JakeLin) [reply IBAnimatable/InteractiveGestureType.swift:25]: Yes, we can trim everything before and the last dot.
